### ISSUE TITLE
stubby.pid is not created as expected

### ISSUE THREAD

**Issue by jdkbx** [getdnsapi/stubby#30]
Something in the configure phase seems broken. I just tried the getdns git with stubby as a submodule and when starting stubby with the -g flag I get the output
`Could not write pid to "@runstatedir@/stubby.pid": No such file or directory`
which seems as if some replacement failed.
When I use the stubby git repository on its own, the stubby.pid file is create at / instead of /var/run/ which also seems to be a replacement problem. Unfortunately I don't know enough about autotools to fix this.

**Comment by saradickinson** [comment]
@wtoorop I see this isssue on macOS too. Seems that runstatedir isn't available in many versions of autoreconf. Do you have a preferred alternative?

**Comment by wtoorop** [comment]
I read it is only available from autoconf 2.70.  Strangely enough this is not yet released, however my (2.69) autoconf does support it.  We could replace it with @localstatedir@/run for the time being (as suggested in the doc: https://www.gnu.org/prep/standards/html_node/Directory-Variables.html )
And add AC_PREREQ([2.70]) to the top of configure.ac once it is released.

Or maybe introduce a new variable which is initialized to @localstatedir@/run , but which can be overridden by an option.  I notice that the @runstatedir@ on my ubuntu 17.10 system is /run (which is different from @localstatedir@/run ) so at least debian/ubuntu packagers might appreciate such an option.

**Comment by saradickinson** [comment]
Indeed, although it seems that support for it was back-ported into debian autotools. 

I think Unbound does the latter of your two suggestions. Will take a look.

**Comment by saradickinson** [comment]
PR #42 should fix this in Stubby. There is a corresponding fix in getdns

**Comment by jdkbx** [comment]
Thanks for your work.
The default now seem to be /var/lib/run, which is uncommon on my system, but I can override it. At the same time that showed a different problem along issue#26 as $(DESTDIR) should be respected in Makefile.am when creating that directory. Without this I do get sandbox violations with gentoos package management.

**Comment by ArchangeGabriel** [comment]
It seems to be `/usr/var/run` here:
```
/usr/bin/install -c -m 755 -d /usr/var/run
/usr/bin/install: cannot create directory ‘/usr/var’: Permission denied
```
And this line corroborates this:
```
  --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
```
But it also lacks the `$(DESTDIR)` indeed.

That being said, it will ends being and empty dir and thus removed by our build system. But since `/var/run` (the correct value) already exists on the system (and by the way it’s a symlink to `/run`), that’s not an issue.

**Comment by saradickinson** [comment]
So we default the piddir to the following:
```default_piddir=${runstatedir:-"${localstatedir}/run"}```
where according to the GNU reference above:
```$(localstatedir) should normally be /usr/local/var, but write it as $(prefix)/var.```
So this can be system dependant. 

Sorry about the issue with missing DESTDIR - fix is in #43 and https://github.com/getdnsapi/getdns/pull/355

**Comment by jdkbx** [comment]
Coming to think of it, the whole line in the Makefile.am that creates the directory seems redundant. But I am not completely sure. It seems to only makes sense when you compile locally and run stubby manually.
I just made a gentoo ebuild (install script for package management) for stubby. When I start stubby through systemd, systemd-tmpfiles creates the directory as specified in the conf file under systemd. For openrc based systems I wrote an init script that takes care of creating and chown-ing the directory.

I do not know, how other software handles this. Maybe I should take a look.
